group-coordinator/src/main/java/org/apache/kafka/coordinator/group/Group.java

```diff
@@ -52,6 +53,21 @@ public String toString() {
         private final static Map<String, GroupType> NAME_TO_ENUM = Arrays.stream(values())
             .collect(Collectors.toMap(type -> type.name.toLowerCase(Locale.ROOT), Function.identity()));
 
+        public static GroupType forId(int id) {
+            switch (id) {
+                case 0:
+                    return CONSUMER;
+                case 1:
+                    return CLASSIC;
+                case 2:
+                    return UNKNOWN;
```

Comment:
Query: Shouldn't we maintain unknown either the last or first?

Reply:
It's an existing enum. I was nervous of changing it.

Reply:
Hmmm, I thought so but doesn't seems right for unknown though. May be it should be the first one or a higher value so groups can grow.

Reply:
Not sure how it could affect, need to look into